lib/bootloader_setup.pm

```diff
@@ -1246,7 +1246,7 @@ Returns the array of the boot parameters.
 =cut
 
 sub parse_bootparams_in_serial {
-    my $parsed_string = wait_serial(qr/command line:.*/msi);
+    my $parsed_string = wait_serial(qr/command line:.*\[/msi);
```

Comment:
I've re-triggered a run here: https://openqa.suse.de/tests/5456257#step/bootloader_start/11 seems that now we do a bit of an overkill, as grab in greedy way, as we have `[` at the beginning of each line. Could you try non-greedy regexp, AFAIR shoud be `.*?` instead of just `.*`. As `[` obviously matches any symbol.

Reply:
I can add it just to be more strict . The initial regex was as greedy, before this modification. See original output https://openqa.suse.de/tests/5443086#step/bootloader_start/11. But as far as I understand, the regex in wait_serial is supposed to tell the serial "wait until we find this regex". In case of "/command line:.*\[/msi" the first line after boot parameters, will send the signal that it's found and that might be the reason for the sporadic failures we had before. I am not sure why the output is so long in "step/bootloader_start/11" but I would assume it has to do with the logs being too fast. See output after changing regex to "/command line:.*?\[/msi" for x86 : http://falafel.suse.cz/tests/1215#step/bootloader/7
It doesn't seem to change anything. Did I misunderstand something?

Reply:
Yeah, looks like there is no difference in behavior here, so my assumption was wrong. Let's leave it like it is. I can see that wait_serial returns in 2 seconds which is good enough for our case. Sorry for the confusion.